src/app/dataset-block/metadata-block/components/event-details/builder.events.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import { BasePropertyComponent } from "./components/common/base-property/base-property.component";
+import { MetadataEvent } from "../../../../api/kamu.graphql.interface";
+import { SetPollingSource } from "src/app/api/kamu.graphql.interface";
+import { Type } from "@angular/core";
+import { SET_POLLING_SOURCE_DESCRIPTORS } from "./components/set-polling-source-event/set-polling-source-event.source";
+
+export interface EventRow {
+    label: string;
+    tooltip: string;
+    value?: unknown;
+    separateRowForValue?: boolean;
+    presentationComponent: Type<BasePropertyComponent>;
+}
+
+export interface EventSection {
+    title: string;
+    rows: EventRow[];
+}
+
+enum EventTypes {
+    SetPollingSource = "SetPollingSource",
+}
+
+enum SetPollingSourceProperty {
+    READ = "read",
+    FETCH = "fetch",
+    MERGE = "merge",
+    PREPROCESS = "preprocess",
+    PREPARE = "prepare",
+}
+
+const keysForSeparateRow: string[] = ["queries"];
```

Comment:
Make a boolean property in the main descriptors instead, it should be `false` by default.